app/frontend/src/features/profile/ProfileForm.tsx

```diff
@@ -20,11 +22,12 @@ const useStyles = makeStyles({
 
 export default function EditProfileForm() {
   const classes = useStyles();
-  const { authState, profileActions } = useAuthContext();
-  const user = authState.user;
-  const [alertState, setShowAlertState] = useState<
-    "success" | "error" | undefined
-  >();
+  const {
+    updateUserProfile,
+    status: updateStatus,
+    reset: resetUpdate,
+  } = useUpdateUserProfile();
+  const { data: user } = useCurrentUser();
   const [errorMessage, setErrorMessage] = useState<string | null>(null);
```

Comment:
I wonder if there's a chance this could cause the setState on unmounted component error with this. Probably likely on slower connections while updating is happening and user navigates away...